src/extensions/renderer/canvas/drawing-edges.js

```diff
@@ -27,10 +27,11 @@ CRp.drawEdge = function( context, edge, shiftToOriginWithBb, drawLabel ){
   let opacity = edge.pstyle('opacity').value;
   let lineStyle = edge.pstyle('line-style').value;
   let edgeWidth = edge.pstyle('width').pfValue;
+  let edgeCap = edge.pstyle('edge-cap').value;
 
   let drawLine = ( strokeOpacity = opacity ) => {
     context.lineWidth = edgeWidth;
-    context.lineCap = 'butt';
+    context.lineCap = edgeCap;
```

Comment:
The `lineCap` should be reset to `'butt'` at the end of the function.  

Background: We don't use `.save()` or `.restore()`, because they're expensive.  So you have to clean up the canvas state for things like lineCap, transform, etc. manually.